src/ServiceControl.Monitoring/QueueLength/QueueLengthDataStore.cs

```diff
@@ -38,12 +75,30 @@ void UpdateSends(IEnumerable<JToken> sends)
                 var key = tags.GetTagValue("key");
                 var value = send.Value<long>("Count");
 
-                calculator.UpdateSentSequence(key, value);
+                var updatedVirtualQueues = calculator.UpdateSentSequence(key, value);
+
+                allUpdatedVirtualQueues.AddRange(updatedVirtualQueues);
             }
+
+            return allUpdatedVirtualQueues.Select(v => v.EndpointName).Distinct().ToList();
+        }
+
+        void SnapshotValueForEndpoint(string endpointName)
+        {
+            var endpointQueueLength = calculator
```

Comment:
I think I need to get better understanding of this snapshotting approach. Could you describe it?